### PR TITLE
Added wallet_revokePermissions rpc method

### PR DESCRIPTION
**Pull request by shanejonas**
## **Description**
This PR adds `wallet_revokePermissions` rpc method support that was added in `permission-controller/rpc-methods` that went in this PR https://github.com/MetaMask/core/pull/1889.

All that is needed is to add the hook as the rpc-method gets automatically added [here](https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/lib/rpc-method-middleware/createMethodMiddleware.js#L13).

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/1681

## **Manual testing steps**

1. Go to this page https://docs.metamask.io/wallet/reference/wallet_requestpermissions/
2. Hit `Send Request` to request permissions
3. Click Next, Click Connect
4. Go to this page https://docs.metamask.io/wallet/reference/wallet_getpermissions/
5. Click `Send Request` to get the permissions
6. See that there is permissions
7. paste this in the console:
```js
await window.ethereum.request({
  "method": "wallet_revokePermissions",
  "params": [
    {
      "eth_accounts": {}
    }
  ]
});
```
8. Click `Send Request` to get the permissions
9. see that there is no permissions and they have been revoked.

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by BelfordZ — DISMISSED** [review]

File: app/scripts/metamask-controller.js [hunk 1]
```diff
@@ -4529,7 +4529,11 @@ export default class MetamaskController extends EventEmitter {
             this.permissionController,
             { origin },
           ),
-
+        revokePermissionsForOrigin: (permissionKeys) => {
+          this.permissionController.revokePermissions({
+            [origin]: permissionKeys,
+          });
```
Comment (adonesky1) [inline app/scripts/metamask-controller.js:4535]: @shanejonas and I discussed and we may want to change the method signature for the hook so we don't necessarily have to do this formatting here. But for now this works.

**Review by adonesky1 — DISMISSED** [review]
I've tested and it works for me! LGTM. Could we just get a lil PR description

**Comment by adonesky1** [comment]
@metamaskbot update-policies

**Comment by jiexi** [comment]
Tested. Working for me too

**Comment by adonesky1** [comment]
@SocketSecurity ignore @metamask/permission-controller@6.0.0

**Comment by adonesky1** [comment]
@SocketSecurity ignore @metamask/approval-controller@5.0.0

**Review by adonesky1 — APPROVED** [review]

**Review by jiexi — APPROVED** [review]

**Pull request merged**